### ISSUE TITLE
Softmax Docs Example Deprecated

### ISSUE THREAD

**Issue by ZaydH** [pytorch/pytorch#20301]
## 📚 Documentation

In the master docs for `nn.Softmax()`, the example provided is:

        >>> m = nn.Softmax()
        >>> input = torch.randn(2, 3)
        >>> output = m(input)

However, this implementation appears deprecated in 1.1.  You now get a warning like:

`UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.`

If you agree, I can change the example to `m = nn.Softmax(dim=1)` and issue a pull request so there is no warning.

**Version**: I observed the warning with Python 1.1.0 and Python 3.7.1 on OSX.

**Comment by SsnL** [comment]
Please make a PR. Thank you!!

**Comment by ZaydH** [comment]
I modified the docstring directly.  I assume that is all that is typical.  If I should rebuild the docs, etc., please let me know.

**Comment by vadimkantorov** [comment]
A proposal: a magic that would decorate certain examples to run as tests

**Comment by ZaydH** [comment]
I assume these examples are being run through with `doctest` (they have similar format).  Is that not correct?

**Comment by ailzhang** [comment]
Closing as the PR is merged. @vadimkantorov please make a separate issue for running examples code. (I'm not sure how it's implemented, but I guess it's manually pasted there :P
